gpii/node_modules/journal/src/SettingsDir.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * GPII SettingsDir.js
+ *
+ * Determines the user's settings directory on a variety of platforms,
+ * and manages establishment of a subdirectory "/gpii" to hold writable files
+ * for the system's journal and other purposes
+ *
+ * Copyright 2016 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ */
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    os = require("os"),
+    fs = require("fs");
+
+fluid.defaults("gpii.settingsDir", {
+    gradeNames: "fluid.component",
+    members: {
+        gpiiSettingsDir: "@expand:{that}.getGpiiSettingsDir()"
+    },
+    invokers: {
+        getGpiiSettingsDir: "gpii.settingsDir.gpii({that}.getSettingsDir)",
+        getSettingsDir: "gpii.settingsDir.temp"
+    },
+    listeners: {
+        "onCreate.createGpiiDir": "gpii.settingsDir.createGpii"
+    }
+});
+
+gpii.settingsDir.createGpii = function (that) {
+    fluid.log(fluid.logLevel.WARN, "Creating GPII settings directory in ", that.gpiiSettingsDir);
+    try { // See code skeleton in http://stackoverflow.com/questions/13696148/node-js-create-folder-or-use-existing
+        fs.mkdirSync(that.gpiiSettingsDir);
+    } catch (e) {
+        if (e.code !== "EEXIST") {
+            fluid.fail("Unable to create GPII settings directory, code is " + e.code + ": exception ", e);
+        }
+    }
+};
+
+gpii.settingsDir.gpii = function (getSettingsDir) {
+    return getSettingsDir() + "/gpii";
+};
+
+gpii.settingsDir.windows = function () {
+    return process.env.APPDATA;
+};
+
+gpii.settingsDir.standard = function () {
+    return os.homedir();
+};
+
+gpii.settingsDir.temp = function () {
+    return os.tmpdir();
```

Comment:
I'm thinking about the public access, shared machine case. If an OS user account is shared between different people, we would need to make sure that the journal files were cleared out between users. Otherwise, someone could determine the settings used by previous users of the machine by reading the journal files. I imagine cleaning the filesystem is a requirement in general in this shared machine scenario as there are lots of other potentially sensitive files created in a user session (such as browser history) -- but I will leave a note here anyway as I think of it.

In does suggest that we need to store our files in a space that does get cleared out -- such as the user's home directory.